pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -672,8 +672,10 @@ protected CompletableFuture<Void> doReconsumeLater(Message<?> message, AckType a
                         return null;
                     });
                 } else {
-                    TypedMessageBuilder<T> typedMessageBuilderNew = retryLetterProducer.newMessage()
-                            .value(retryMessage.getValue())
+                    assert retryMessage != null;
+                    TypedMessageBuilder<byte[]> typedMessageBuilderNew = retryLetterProducer
+                            .newMessage(Schema.AUTO_PRODUCE_BYTES(message.getReaderSchema().get()))
```

Comment:
This line is copied from how we work for deadletter, I'm wondering whether it can produce NPE at `message.getReaderSchema().get()`. cc @congbobo184

Reply:
Unless the user code like this `new TypedMessageBuilderImpl(null, null)` and this is not the `reconsumeLaster` correct use way.
we use `reconsumeLaster` should use the invoke `consumer.receive()` returned messages,  or construct the correct message yourself

Reply:
Hi @congbobo184 Could you give some advice?

Reply:
@poorbarcode I read https://github.com/apache/pulsar/pull/19051#discussion_r1056922378 is enough for a resolution :)